keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/DeliveryConfigRepositoryTests.kt

```diff
@@ -599,6 +599,29 @@ abstract class DeliveryConfigRepositoryTests<T : DeliveryConfigRepository, R : R
         }
       }
 
+      context("updating an existing delivery config with a new version of a resource") {
+        before {
+          storeArtifacts()
+          storeResources()
+          store()
+
+          val resource = deliveryConfig.environments.first().resources.first { it.kind == parseKind("ec2/cluster@v1") }
+            .run {
+              copy(spec = DummyResourceSpec(id = id, application = application))
+            }
+
+          resourceRepository.store(resource)
+        }
+
+        test("retrieving the delivery config includes the new version of the resource") {
+          getByName()
+            .isSuccess()
+            .get { resources.map(Resource<*>::version) }
+            .hasSize(deliveryConfig.resources.size)
+            .contains(2)
+        }
+      }
```

Comment:
This tests that when we create a v n+1 of a resource, the foreign key linking it to the environment updates so that the environment now contains the new resource version

Reply:
Is there a chance this might break code that was previously using `deliveryConfig.resources` or `environment.resources` to loop through resources, assuming there would be a single object for each resource ID?

Reply:
For example, when generating resource summaries for the UI:
https://github.com/spinnaker/keel/blob/37ea959427af8fa2c2e9ce2d39d2f07f3c9ac3d5/keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt#L184-L186

Reply:
> (I'm assuming we now have all versions of each resource in that list)

We don't. The environment will only contain the latest version of the resource. In the code where the new version of the resource is persisted I break the existing `environment_resource` link and replace it with a new one.